### ISSUE TITLE
[no-mixed-types] Doesn't catch mixed types in Readonly

### ISSUE THREAD

**Issue by RebeccaStevens** [eslint-functional/eslint-plugin-functional#500]
```ts
type Foo = Readonly<{
  prop: number;
  method: () => void;
}>
```
